components/GovernancePower/Vanilla/useDepositCallback.tsx

```diff
@@ -38,24 +38,53 @@ export const useDepositCallback = (
         ASSOCIATED_TOKEN_PROGRAM_ID,
         TOKEN_PROGRAM_ID,
         mint,
-        walletPk // owner
+        walletPk, // owner
+        true
       )
 
       const instructions: TransactionInstruction[] = []
       const signers: Keypair[] = []
 
-      const transferAuthority = approveTokenTransfer(
+      let transferAuthority;
+      console.log(wallet?.name)
+      if (wallet?.name == "SquadsX") {
+      transferAuthority = approveTokenTransfer(
         instructions,
         [],
         userAtaPk,
         wallet!.publicKey!,
-        amount
-      )
+        amount,
+        true,
+        wallet.publicKey!
+      ) 
+      } else {
+      transferAuthority = approveTokenTransfer(
+          instructions,
+          [],
+          userAtaPk,
+          wallet!.publicKey!,
+          amount
+        ) 
 
       signers.push(transferAuthority)
+      }
 
       const programVersion = await fetchProgramVersion(connection, realm.owner)
 
+      if (wallet?.name == "SquadsX") {
```

Comment:
could you explain this and why a special case is needed? and if it really is needed, please leave comments in the code about it

Reply:
Added comments.